lib/merchant_repository.rb

```diff
@@ -1,48 +1,69 @@
+require_relative "merchant"
 require "csv"
-require "./lib/merchant"
 
 class MerchantRepository
   attr_reader :filename,
               :merchants
-  def initialize(filename)
+  def initialize(filename) #parent)
     @filename = filename
+    #@parent = parent
     @merchants = Array.new
     generate_merchants(filename)
   end
 
   def generate_merchants(filename)
     merchants = CSV.open filename, headers: true, header_converters: :symbol
     merchants.each do |row|
-      @merchants << Merchant.new(row[:id], row[:name])
+      @merchants << Merchant.new(row[:id], row[:name])#, self)
     end
   end
 
+  def all
+    @merchants
+  end
+
   def find_by_id(id)
-    merchants.find_all do |merchant|
-      merchant[:id] == id
+     id_found = merchants.find do |merchant|
+      merchant.id == id
     end
   end
 
   def find_by_name(name)
-    merchants.find do |merchant|
-      merchant[:name].downcase == name.downcase
+    name_found = merchants.find do |merchant|
+      merchant.name.downcase == name.downcase
     end
   end
 
   def find_all_by_name(name)
-    found_merchants = merchants.find do |merchant|
-      merchant[:name].downcase == name.downcase
+    found_merchants = merchants.find_all do |merchant|
+      merchant.name.downcase.include?(name.downcase)
+    end
+  end
+
+  def highest_merchant_id_plus_one
+    highest = @merchants.max do |merchant|
+      merchant.id
     end
+    highest.id + 1
   end
 
-  def create(attributes)
+  def create(hash)
+    new_merchant = Merchant.new(highest_merchant_id_plus_one, hash[:name])
+    @merchants << new_merchant
+    new_merchant
   end
 
-  def update(id, attributes)
+  def update(id, name_hash)
+    update_merchant = find_by_id(id)
+    update_merchant.update(name_hash[:name]) if !name_hash[:name].nil?
```

Comment:
Awesome logic! If the name is not nil then update it! this sure handles calling something on nil